c10/cuda/CUDAMacros.h

```diff
@@ -47,5 +49,6 @@ o */
 // fbcode depends on this value being 16
 #define C10_COMPILE_TIME_MAX_GPUS 16
 #else
-#define C10_COMPILE_TIME_MAX_GPUS 120
+constexpr std::int64_t C10_COMPILE_TIME_MAX_GPUS =
+    std::numeric_limits<int8_t>::max() + 1;
```

Comment:
You do `<=` checks on this number below, so the +1 is not ok!

Reply:
+1 for device 0